### PR TITLE
Add support for refreshing Azure tokens.

### PR DESCRIPTION
**Pull request by brendandburns**
@roycaihw 

This requires an update to the 'requirements.txt' to import the `adal` library, but it appears that is in a different repo?

@roycaihw is there advice for how to add dependencies?

Thanks!

### CODE REVIEW

**Comment by roycaihw** [comment]
We use [tox](https://github.com/kubernetes-client/python-base/blob/master/run_tox.sh) to test `python-base` change with `python` HEAD. I think you could add the dependency to [python/test-requirements.txt](https://github.com/kubernetes-client/python/blob/master/test-requirements.txt) first to make the CI passes. And after this PR is merged, when we bump the `python-base` submodule in `python` we can sync the actual [requirements.txt](https://github.com/kubernetes-client/python/blob/master/requirements.txt).

**Comment by brendandburns** [comment]
Thanks, filed https://github.com/kubernetes-client/python/pull/579 for requirements.txt

File: config/kube_config.py [hunk 1]
```diff
@@ -202,10 +204,29 @@ def _load_azure_token(self, provider):
...
+        tenant = config['tenant-id']
+        authority = 'https://login.microsoftonline.com/{}'.format(tenant)
+        context = adal.AuthenticationContext(
+            authority, validate_authority=True,
+        )
+        refresh_token = config['refresh-token']
+        client_id = config['client-id']
+        token_response = context.acquire_token_with_refresh_token(
+            refresh_token, client_id, '00000002-0000-0000-c000-000000000000')
+
```
Comment (roycaihw) [inline config/kube_config.py]: the CI pycodestyle complains about blank line containing whitespace

**Pull request merged**